research/delf/delf/python/datasets/revisited_op/dataset.py

```diff
@@ -458,12 +462,83 @@ def ReadMetricsFile(metrics_path):
 
     # Parse mean precisions.
     mean_precisions[protocol] = np.array(
-        _ParsePrScores(file_contents_stripped[i + 2], len(pr_ranks)),
-        dtype=float) / 100.0
+      _ParsePrScores(file_contents_stripped[i + 2], len(pr_ranks)),
+      dtype=float) / 100.0
 
     # Parse mean recalls.
     mean_recalls[protocol] = np.array(
-        _ParsePrScores(file_contents_stripped[i + 3], len(pr_ranks)),
-        dtype=float) / 100.0
+      _ParsePrScores(file_contents_stripped[i + 3], len(pr_ranks)),
+      dtype=float) / 100.0
 
   return mean_average_precision, pr_ranks, mean_precisions, mean_recalls
+
+
+def configdataset(dataset, dir_main):
+  """Configures the paths to the test dataset files.
+
+  Args:
+    dataset: String, dataset name: either 'roxford5k' or 'rparis6k'.
+    dir_main: String, path to the folder containing ground truth files.
+
+  Returns:
+    cfg: Dataset configuration.
+
+  Raises:
+    ValueError: If an unknown dataset name is provided as an argument.
+  """
+  DATASETS = ['roxford5k', 'rparis6k']
+  dataset = dataset.lower()
+
+  if dataset not in DATASETS:
+    raise ValueError('Unknown dataset: {}!'.format(dataset))
+
+  # Loading imlist, qimlist, and gnd in configuaration as a dictionary.
+  gnd_fname = os.path.join(dir_main, 'gnd_{}.pkl'.format(dataset))
+  with tf.io.gfile.GFile(gnd_fname, 'rb') as f:
+    cfg = pickle.load(f)
+  cfg['gnd_fname'] = gnd_fname
+  if dataset == 'rparis6k':
+    dir_images = 'paris6k_images'
+  elif dataset == 'roxford5k':
+    dir_images = 'oxford5k_images'
+
+  cfg['ext'] = '.jpg'
+  cfg['qext'] = '.jpg'
+  cfg['dir_data'] = os.path.join(dir_main)
+  cfg['dir_images'] = os.path.join(cfg['dir_data'], dir_images)
+
+  cfg['n'] = len(cfg['imlist'])
+  cfg['nq'] = len(cfg['qimlist'])
+
+  cfg['im_fname'] = config_imname
```

Comment:
IIUC, this is adding the function to the output dict. To simplify things, we could do the following:
cfg['im_fname'] = lambda cfg, i: os.path.join(cfg['dir_images'], cfg['imlist'][i] + cfg['ext'])

If this fits in one line, that would work (https://google.github.io/styleguide/pyguide.html#210-lambda-functions). Otherwise, if `config_imname` is not needed outside of this file, just add it as an internal function to the `configdataset` function.

Same comments for cfg['qim_fname'].